pvlib/spectrum/mismatch.py

```diff
@@ -325,12 +327,20 @@ def spectral_factor_firstsolar(precipitable_water, airmass_absolute,
     min_precipitable_water : float, default 0.1
         minimum atmospheric precipitable water. Any ``precipitable_water``
         value lower than ``min_precipitable_water``
-        is set to ``min_precipitable_water`` to avoid model divergence. [cm]
+        is set to ``min_precipitable_water``. [cm]
 
     max_precipitable_water : float, default 8
         maximum atmospheric precipitable water. Any ``precipitable_water``
         value greater than ``max_precipitable_water``
-        is set to ``np.nan`` to avoid model divergence. [cm]
+        is set to ``np.nan``. [cm]
```

Comment:
Do you think it is important for large/min values to be handled consistently? Since it seems agreed in #2086 that defaulting to the maximum is the right choice for the new airmass parameters, I was wondering whether the `np.nan` default for max pw should be changed too, unless there is a reason for replacing with `np.nan` rather than `max_precipitable_water` in this specific case?
But yes if the change made in the `mismatch.py` calculation is agreed then I will update the docs here as you have noted, thanks for picking up on that

Reply:
In the case of `pw`, `pw>8cm` is physically improbable and I think the model authors replaced with `np.nan` because `pw>8` indicated erroneous data. I have a mild preference for retaining this behavior.

In the case of `AM`, `AM>10` indicates a sun position right on the geometric horizon, so I have difficulty thinking that any model result with `AM>10` could be meaningful. Since `AM` is a constructed rather than measured value, replacing `AM>10` with 10 rather than `np.nan` seems OK to me.

Reply:
Makes sense, thanks for the clear explanation @cwhanse